tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/Shared/SpanPointers.cs

```diff
@@ -0,0 +1,121 @@
+// <copyright file="SpanPointers.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AWS.Shared;
+
+/// <summary>
+///  SpanPointer helper methods
+/// </summary>
+internal static class SpanPointers
+{
+    // The pointer direction will always be down. The serverless agent handles cases where the
+    // direction is up.
+    private const string DownDirection = "d";
+    private const string LinkKind = "span-pointer";
+    private const int SpanPointerHashSizeBytes = 16;
+    private const string S3PtrKind = "aws.s3.object";
+
+    // S3 hashing rules: https://github.com/DataDog/dd-span-pointer-rules/blob/main/AWS/S3/Object/README.md
+    public static void AddS3SpanPointer(Span span, string bucketName, string key, string eTag)
+    {
+        var hash = GeneratePointerHash(bucketName, key, StripQuotes(eTag));
+        var spanLinkAttributes = new List<KeyValuePair<string, string>>(4)
+        {
+            new("ptr.kind", S3PtrKind),
+            new("ptr.dir", DownDirection),
+            new("ptr.hash", hash),
+            new("link.kind", LinkKind)
+        };
+
+        var spanLink = new SpanLink(SpanContext.ZeroContext, spanLinkAttributes);
+        span.AddLink(spanLink);
+    }
+
+    // Hashing rules: https://github.com/DataDog/dd-span-pointer-rules/tree/main?tab=readme-ov-file#general-hashing-rules
+    private static string GeneratePointerHash(params string[] components)
+    {
+#if NETCOREAPP3_0_OR_GREATER || NET5_0_OR_GREATER || NETSTANDARD2_1_OR_GREATER
```

Comment:
This is enough and will make it easier to find later if we ever need to change it (because it's what we use everywhere).
```suggestion
#if NETCOREAPP3_1_OR_GREATER
```